pyiceberg/table/__init__.py

```diff
@@ -350,6 +357,241 @@ class RemovePropertiesUpdate(TableUpdate):
     removals: List[str]
 
 
+class TableMetadataUpdateContext:
```

Comment:
I think we can mark this one private for the module:
```suggestion
class _TableMetadataUpdateContext:
```